### PR TITLE
feat(FX-3898): preselects the existing "Recently Added" sort option for `artsy-2` partner

### PR DESCRIPTION
**Pull request by dimatretyak**
The type of this PR is: **Feat**

This PR solves [FX-3898]

artsy-2 partner: https://recently-added-sort.artsy.net/partner/artsy-2/works
Other partner: https://recently-added-sort.artsy.net/partner/white-cube/works

### Description
We need to preselect the existing "Recently Added" sort option only for `artsy-2` partner

[FX-3898]: https://artsyproduct.atlassian.net/browse/FX-3898?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by nickskalkin — APPROVED** [review]
great! 👍

**Review by mdole — APPROVED** [review]
nice work! ship it!

File: src/Apps/Partner/Routes/Works/index.tsx [hunk 1]
```diff
@@ -9,25 +9,30 @@ import {
...
   partner,
   relay,
-  match,
 }) => {
   const { sidebar } = partner
+  const { match } = useRouter()
+  const filters = match?.location?.query ?? {}
+
+  // Preselects "Recently added" sort option for artsy-2 partner by default
+  if (match?.params?.partnerId === "artsy-2" && filters.sort === undefined) {
```
Comment (dblandin) [inline src/Apps/Partner/Routes/Works/index.tsx:29]: **question / non-blocking**: Would it be worth pulling this slug into a config variable? I remember discussing whether we can reclaim the `artsy` partner slug, at which point we would need to update any hardcoded values.

Maybe something like `ARTSY_MERCHANDISING_PARTNER_SLUGS=artsy-2` and we can later switch to include future slugs which should gain the same behavior: `ARTSY_MERCHANDISING_PARTNER_SLUGS=artsy-2,artsy,artsy-3`

**Review by dblandin — DISMISSED** [review]
Looks good to me! One question about pulling the slugs into the config which we can either investigate within this PR or afterwards.

File: src/Apps/Partner/Routes/Works/index.tsx (hunk 1 above)
Comment (dimatretyak) [reply src/Apps/Partner/Routes/Works/index.tsx:29]: yep, sure. I will add the necessary changes

Comment (dblandin) [reply src/Apps/Partner/Routes/Works/index.tsx:29]: One update: the [Artsy MSO account](https://www.artsy.net/partner/artsy) now has the `artsy` slug. `artsy-2` and `artsy-mso` will continue to work as previous slugs but the primary is now `/partner/artsy`.

Comment (dimatretyak) [reply src/Apps/Partner/Routes/Works/index.tsx:29]: should we add `artsy-2`, `artsy-mso` to `ARTSY_MERCHANDISING_PARTNER_SLUGS`?

Comment (dblandin) [reply src/Apps/Partner/Routes/Works/index.tsx:29]: Sure, let's go ahead and add those too since they don't redirect.

Comment (dimatretyak) [reply src/Apps/Partner/Routes/Works/index.tsx]: Done

**Review by dblandin — APPROVED** [review]
LGTM 👍🏼

**Review by gkartalis — APPROVED** [review]

**Pull request merged**